src/promo/ThinBanner.jsx

```diff
@@ -0,0 +1,106 @@
+import Button from '../Button'
+import ArrowRightIcon from '@material-ui/icons/KeyboardArrowRight'
+import PropTypes from 'prop-types'
+import React from 'react'
+import withStyles from '@material-ui/core/styles/withStyles'
+import hexRgb from 'hex-rgb'
+
+const styles = theme => ({
+  button: {
+    backgroundColor: theme.palette.core && theme.palette.core.main,
+    borderRadius: 0,
+    color: [theme.palette.primary.contrastText],
+    fontWeight: 'bold',
+    justifyContent: 'left',
+    padding: '9px 12px', // special case, needs to match padding on existing topbar
+
+    '&:hover': {
+      backgroundColor: theme.palette.core && theme.palette.core.main
+    }
+  },
+  icon: {
+    position: 'absolute',
+    right: theme.spacing.unit
+  }
+})
+
+/**
+ * The `<ThinBanner>` component is a clickable block that can
+ * contain a short amount of text.
+ *
+ * Intended for use on mobile as an alternative to `<DonationBanner>`, it can
+ * also be used to promote surveys (traditionally with a green background) and
+ * newsletters (traditionally with a blue background).
+ *
+ * It optionally accepts both backgroundColor and color props. If provided, it'll
+ * use those. Without a backgroundColor, it'll default to TC red.
+ *
+ * Without a color, it'll determine the brightness of the background color, and if
+ * it's darker you'll see white text, if it's lighter you'll see dark grey text.
+ *
+ * ~~~js
+ * <ThinBanner>
+ *   Some nice text
+ * </ThinBanner>
+ * ~~~
+ *
+ */
+export const ThinBanner = ({ children, backgroundColor, color, classes, onClick }) => {
+  let textColor
```

Comment:
Mixing up my other comments, suggestion (if the text color is desired as a prop):

```js
const DARK_TEXT_COLOR = '#383838'
const LIGHT_TEXT_COLOR = '#ffffff'
const LUMINANCE_THRESHOLD = 0.5

const textColor = color ? color :
  perceivedLuminance(backgroundColor) < LUMINANCE_THRESHOLD ? DARK_TEXT_COLOR : LIGHT_TEXT_COLOR
```